korangar/src/loaders/animation/mod.rs

```diff
@@ -0,0 +1,474 @@
+use std::cmp::{max, min};
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use cgmath::Vector2;
+use derive_new::new;
+use korangar_interface::elements::PrototypeElement;
+use num::Zero;
+use wgpu::{Device, Extent3d, Queue, TextureDescriptor, TextureDimension, TextureFormat, TextureUsages};
+
+use super::error::LoadError;
+use crate::graphics::Texture;
+use crate::loaders::{ActionLoader, Actions, AnimationState, Sprite, SpriteLoader};
+use crate::{Color, EntityType};
+
+// TODO: use cache later, the memory will be increase with this hashmap, until
+// the program is out of memory.
+#[derive(new)]
+pub struct AnimationLoader {
+    device: Arc<Device>,
+    queue: Arc<Queue>,
+    #[new(default)]
+    // The string will be type of entity
+    // 0_{body_id}_{head_id}
+    // 1_{monster_id}
+    // 2_{npc_id}
+    cache: HashMap<String, AnimationData>,
+}
+
+impl AnimationLoader {
+    pub fn load(
+        &mut self,
+        sprite_loader: &mut SpriteLoader,
+        action_loader: &mut ActionLoader,
+        entity_filename: Vec<String>,
+        entity_type: EntityType,
+    ) -> Result<AnimationData, LoadError> {
+        // Create animation pair with sprite and action
+        let vec: Vec<AnimationPair> = entity_filename

```

Comment:
"vec" naming here is very confusing. Better find a more descriptive name.

Reply:
Renamed it to animation_pairs